crowbar_framework/app/models/node_object.rb

```diff
@@ -659,7 +659,12 @@ def rebuild_run_list
     crowbar["run_list_map"] = {} if crowbar["run_list_map"].nil?
 
     # Cull by state
-    map = crowbar["run_list_map"].select { |k,v| v["states"].include?("all") or v["states"].include?(self.crowbar['state']) }
+    map = crowbar["run_list_map"].select do |k, v|
+      v["states"].include?(state) ||
+      # When node is in upgrade state, allow only upgrade specific recipes
+      (v["states"].include?("all") && !state.include?("upgrade"))
```

Comment:
Yes, that's just for potential other `_upgrade` state. But better to leave the only name we currently have, I guess...